### PR TITLE
Updates to admin approval form

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Updates to the admin approval form (see [spec](https://docs.google.com/document/d/1uFHbZuP7xkNLIOG4FHNouTdYmnCuEB3JbG7SMC_XRj8/edit)). Most of the edits were deleting questions, hence the largely negative diff.

Not sure if it's helpful, but I did a quick screen capture of the form:

https://github.com/code-dot-org/code-dot-org/assets/46464143/7604a4e1-235b-42a3-bd5a-661fe3c05902

### CODE REVIEW

File: apps/src/code-studio/pd/application/principalApproval/PrincipalApprovalComponent.jsx [hunk 1]
```diff
@@ -375,7 +313,6 @@ PrincipalApprovalComponent.propTypes = {
 PrincipalApprovalComponent.associatedFields = [
   ...Object.keys(PageLabels),
   'doYouApprove',
-  'committedToMasterSchedule',
   'contactInvoicing',
   'contactInvoicingDetail',
```
Comment (TurnerRiley) [inline apps/src/code-studio/pd/application/principalApproval/PrincipalApprovalComponent.jsx]: Should these be deleted too since their fields were deleted above?

File: apps/test/unit/code-studio/pd/application/PrincipalApprovalTest.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ describe('PrincipalApproval', () => {
       PrincipalApprovalComponent.getDynamicallyRequiredFields({
         doYouApprove: 'No',
       })
-    ).to.deep.equal(ALWAYS_REQUIRED_FIELDS);
+    );
```
Comment (TurnerRiley) [inline apps/test/unit/code-studio/pd/application/PrincipalApprovalTest.js]: Sorry if this is part of the spec that I didn't notice, but why is this line changed?

File: dashboard/app/models/pd/application/principal_approval_application.rb [hunk 3]
```diff
@@ -160,8 +140,6 @@ def self.filtered_labels(course, status = 'unreviewed')
 
     def additional_text_fields
       [
-        [:committed_to_master_schedule],
-        [:replace_course, TEXT_FIELDS[:dont_know_explain], :replace_course_other],
         [:do_you_approve],
         [:contact_invoicing],
         [:contact_invoicing_detail]
```
Comment (TurnerRiley) [inline dashboard/app/models/pd/application/principal_approval_application.rb]: Same question here for these two fields as above?

File: dashboard/app/models/pd/application/teacher_application.rb [hunk 4]
```diff
@@ -1170,7 +1161,6 @@ def on_successful_principal_approval_create(principal_approval)
           principal_other_percent: principal_response[:other] ? format("%0.02f%%", principal_response[:other]) : nil,
           principal_wont_replace_existing_course: replace_course_string,
           principal_send_ap_scores: principal_response[:send_ap_scores],
-          principal_pay_fee: principal_response[:pay_fee],
           principal_contact_invoicing: principal_response[:contact_invoicing],
           principal_contact_invoicing_detail: principal_response[:contact_invoicing_detail]
```
Comment (TurnerRiley) [inline dashboard/app/models/pd/application/teacher_application.rb]: Same question with these two fields

File: dashboard/test/controllers/api/v1/pd/application/principal_approval_applications_controller_test.rb [hunk 5]
```diff
@@ -90,8 +87,6 @@ class PrincipalApprovalApplicationsControllerTest < ::ActionController::TestCase
         form_data: build(PRINCIPAL_APPROVAL_HASH_FACTORY,
           do_you_approve: "Other:",
           do_you_approve_other: "this is the other for do you approve",
-          committed_to_master_schedule: "Other:",
-          committed_to_master_schedule_other: "this is the other for master schedule",
           replace_course: "I don't know (Please Explain):",
           replace_course_other: "this is the other for replace course"
```
Comment (TurnerRiley) [inline dashboard/test/controllers/api/v1/pd/application/principal_approval_applications_controller_test.rb]: Wasn't this question removed from the form?

File: dashboard/test/controllers/api/v1/pd/application/principal_approval_applications_controller_test.rb [hunk 6]
```diff
@@ -104,7 +99,6 @@ class PrincipalApprovalApplicationsControllerTest < ::ActionController::TestCase
 
       expected_principal_fields = {
         principal_approval: "Other: this is the other for do you approve",
-        principal_schedule_confirmed: "Other: this is the other for master schedule",
         principal_wont_replace_existing_course: "I don't know (Please Explain): this is the other for replace course",
```
Comment (TurnerRiley) [inline dashboard/test/controllers/api/v1/pd/application/principal_approval_applications_controller_test.rb]: Wasn't this question also removed from the form?

File: dashboard/test/controllers/api/v1/pd/applications_controller_test.rb [hunk 7]
```diff
@@ -780,8 +780,6 @@ class ApplicationsControllerTest < ::ActionController::TestCase
         "Percent of student enrollment by race - Native Hawaiian or other Pacific Islander",
         "Percent of student enrollment by race - American Indian or Native Alaskan",
         "Percent of student enrollment by race - Other",
-        "Are you committed to including this course on the master schedule in #{APPLICATION_CURRENT_YEAR} if this teacher is accepted into the program?",
-        "Will this course replace an existing computer science course in the master schedule? (Principal's response)",
         "How will you implement CS Principles at your school?",
         "If there is a fee for the program, will your teacher or your school be able to pay for the fee?",
```
Comment (TurnerRiley) [inline dashboard/test/controllers/api/v1/pd/applications_controller_test.rb]: I think line 784 can be removed too!

Comment (TurnerRiley) [reply dashboard/test/controllers/api/v1/pd/applications_controller_test.rb]: Same for the invoicing strings below on lines 786-787

File: lib/cdo/shared_constants/pd/principal_approval_application_constants.rb [hunk 8]
```diff
@@ -30,9 +30,6 @@ def self.clean_multiline(string)
       pacific_islander: 'Native Hawaiian or other Pacific Islander',
       american_indian: 'American Indian or Native Alaskan',
       other: 'Other',
-      replace_course: 'Will this course replace an existing computer science course in the master schedule?',
-      understand_fee: 'By checking this box, you indicate that you understand there may be a fee for the professional learning program your teacher attends.',
-      pay_fee: 'If there is a fee for the program, would your school be able to pay for the fee?',
       contact_invoicing: "Contact name for invoicing (if applicable)",
       contact_invoicing_detail: "Contact email or phone number for invoicing (if applicable)",
```
Comment (TurnerRiley) [inline lib/cdo/shared_constants/pd/principal_approval_application_constants.rb]: If the contact invoicing questions are being removed, we can remove these too I think!

**Review by TurnerRiley — COMMENTED** [review]
Left a few questions and some places where I think you can remove more lines!

File: apps/src/code-studio/pd/application/principalApproval/PrincipalApprovalComponent.jsx (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/code-studio/pd/application/principalApproval/PrincipalApprovalComponent.jsx]: Good catch!

File: apps/test/unit/code-studio/pd/application/PrincipalApprovalTest.js (hunk 2 above)
Comment (bethanyaconnor) [reply apps/test/unit/code-studio/pd/application/PrincipalApprovalTest.js]: that is a good question lol. I think there should be something here, even if it's comparing to something different

File: dashboard/app/models/pd/application/principal_approval_application.rb (hunk 3 above)
Comment (bethanyaconnor) [reply dashboard/app/models/pd/application/principal_approval_application.rb]: Removed!

File: dashboard/app/models/pd/application/teacher_application.rb (hunk 4 above)
Comment (bethanyaconnor) [reply dashboard/app/models/pd/application/teacher_application.rb]: Removed! Also removed a couple other fields in this hash

File: apps/test/unit/code-studio/pd/application/PrincipalApprovalTest.js (hunk 2 above)
Comment (bethanyaconnor) [reply apps/test/unit/code-studio/pd/application/PrincipalApprovalTest.js]: It also passes when I add it back 🤷‍♀️

File: dashboard/test/controllers/api/v1/pd/application/principal_approval_applications_controller_test.rb (hunk 5 above)
Comment (bethanyaconnor) [reply dashboard/test/controllers/api/v1/pd/application/principal_approval_applications_controller_test.rb]: Yep -- removed. Also just removed the whole test above, `application update contains replaced courses`

File: apps/test/unit/code-studio/pd/application/PrincipalApprovalTest.js (hunk 2 above)
Comment (TurnerRiley) [reply apps/test/unit/code-studio/pd/application/PrincipalApprovalTest.js]: Does that mean it's succeeding either way but for different reasons? Like without removing the line works because the `expects` statement succeeds, and with removing the line it succeeds since no errors were raised during the test. Or does it seem like the test isn't functioning properly?

Comment (bethanyaconnor) [reply apps/test/unit/code-studio/pd/application/PrincipalApprovalTest.js]: It is succeeding either way but when this line was removed, it wasn't actually testing anything so it was kind of succeeding by default. I'm honestly not sure why this line was removed and it's possible I didn't mean to.

Comment (TurnerRiley) [reply apps/test/unit/code-studio/pd/application/PrincipalApprovalTest.js]: Oh gotcha gotcha!

**Review by TurnerRiley — APPROVED** [review]
